textprocessors.cpp

```diff
@@ -28,6 +28,23 @@ void capitalize_letters(char *text) {
 
 }
 
-void search_replace(char *text, char *search, char *replace) {
-
+void search_replace(string &text, string &search, string &replace) {
```

Comment:
```suggestion
void search_replace(string &text, const string search, const string replace) {
```
Drop the pass-by-reference for `string search, string replace` and make them const.
I know Clang-tidy is saying to use reference, but you can no longer pass a raw string to the function if you pass by reference